### DISCUSSION TITLE
buying advice for motion sensors

### DISCUSSION THREAD

**Discussion by olierxleben** [microsoft/devicescript#465] [Q&A] [answered]
Hi all, 

I want to get startet with Device Script for tinkering with Motion Sensors. I am kind of newbie to the electronics stuff. I guess a Rpi Pico is suitable for that. Now I need a bit of guidance which motion extension board I should buy. Any concerns working with DeviceScript here? 

Thanks in advance, 

Oli

**Comment by pelikhan** [comment]
We are not familiar with motion extension boards. Do you have a link?

You'll want to get ESP32 (esp32-c3 best) if you want to use network connectivity. https://microsoft.github.io/devicescript/devices

**Reply by olierxleben** [reply]
Thanks for your reply. 

I thought about those guys:
<img width="269" alt="image" src="https://github.com/microsoft/devicescript/assets/1403225/16138f32-8682-4cb4-ab10-032f5b30774d">
HC-SR501 and/or HC-SR505

Here are some links/ guides I would like to try out. 

https://www.tomshardware.com/how-to/raspberry-pi-pico-motion-sensor
https://www.freva.com/pir-motion-sensor-on-a-raspberry-pi-pico/

**Reply by olierxleben** [reply]
I also found this one (ESP32): https://www.youtube.com/watch?v=eUtSaV774to

**Reply by pelikhan** [reply]
@mmoskal what hardware motion sensor does this driver apply to?

https://microsoft.github.io/devicescript/api/servers/motion

**Reply by mmoskal** [reply, marked as answer]
Just added the following to the docs: This is typically used with a PIR-based motion detectors, that just output the debounced motion reading as high-level on a given pin. Typical hardware includes: `AM312`, `SR602`.

The ones you're showing will probably also work. Feel free to try them and report back!
